### DISCUSSION TITLE
Add in a zip like format for rulesets that require multiple .dll files.

### DISCUSSION THREAD

**Discussion by juney2008** [ppy/osu#30207] [Ideas]
So I was thinking about an idea of a file format to keep the rulesets folder a bit clean and easier to find what ruleset you're looking for if a ruleset has multiple .dll files that you would need to add to the rulesets folder like "Karaoke" and "IGPlayer".

I'm not sure if this would cause conflicts with compiling the dll files itself so if it does maybe just letting the ruleset folder detect folders would work.

**Comment by peppy** [comment]
Can you give an example of such a ruleset?

**Reply by juney2008** [reply]
Sorry for the very late response since I didn't see this but I can bring out the [karaoke ruleset](https://github.com/karaoke-dev/karaoke) as an example since that uses a lot of dll files in order to run the ruleset.
![image](https://github.com/user-attachments/assets/c0e3ceef-6d61-48a3-9738-20587f658156)

**Reply by peppy** [reply]
One path forward is that rulesets can combine the required assemblies into themselves as a single DLL (I think that's still possible using dotnet tools?)

**Reply by juney2008** [reply]
Ah, I guess that would work.
